src/Monolog/LogProcessor.php

```diff
@@ -14,24 +14,16 @@
 
 class LogProcessor
 {
-    /**
-     * @var AbstractUploader
-     */
+    /** @var null|false|AbstractUploader */
```

Comment:
na co ten false?

Reply:
to je takova vojebavka, protoze ten uploader se tu lazy-inicializuje, takze potrebujes rozlisit 3 stavy:
* nezinicializovany (`null`)
* zinizializovany, ale pro danou konfiguraci uploader neni (`false`)
* zinicializovany a platny uplaoder (`AbstractUploader`)

Druha moznost byla si na do pridat jeste druhou property (ala `$uploaderInitialized`), ale byl jsem linej.

Reply:
Otazka je k čemu je tam teď dobrá ta lazy inicalizace - co by se stalo kdyby tam nebyla?  Resp. důležitý by mi přišlo, aby to vracelo pořád stejnou instanci, ale jestli to bude opakovaně tou factory mi nepřijde jako problém

Reply:
Ma porad stejnou funkci jako do ted a verim, ze kdyz ji tam kdysi nekdo udelal, vedel proc.. nevim co vic z toho vyspekulovat :slightly_smiling_face: Ta factory se stale zavola je 1x, nevola se opakovane (prave diky `null` vs. `false`). Jediny, co se zmenilo, ze `$this->uploader` muze byt nove prazdny i po inicializaci.

Netvrdim, ze se mi to libi :slightly_smiling_face: ale snazil jsme se to udelat v duchu, jak to bylo do ted, s minimalnim mnozstvim zmen. Pokud bychom to chteli udelat poradne, upload erroru by ve finale vubec nemel byt v log processoru, ale mel by to byt samostatny handler.

Reply:
No podle mě byla lazy protože to tady vyhazovalo výjimku https://github.com/keboola/api-error-control/pull/35/files#diff-bf2126e43a648bbf237b047bd68bb4d0cc5acd9dbf04cbe3dc0ca386c9458d3dL66 
Další důvod mohl být aby se to náhodou v constructoru uploaderu někam nepřipojovalo, ale to by se a) stejně nezměnilo a b) afaik stejně neděje